File: ingest/ingest/models/base_station.py
from ingest.models.base import Base
from sqlalchemy import Column, Integer, String, Float, DateTime

class BaseStation(Base):
    __tablename__ = 'base_station'
    
    id = Column(Integer, primary_key=True, autoincrement=False)
    name = Column(String, nullable=False)
    latitude = Column(Float, nullable=False)
    longitude = Column(Float, nullable=False)
    last_ping = Column(DateTime, nullable=False)
    created_at = Column(DateTime, nullable=False)

File: web/src/models/base-station.ts
export interface BaseStation {
  id: number;
  name: string;
  latitude: number;
  longitude: number;
  last_ping: Date;
  created_at: Date;
}

export const initBaseStation = (): BaseStation => ({
  id: 0,
  name: "",
  latitude: 0,
  longitude: 0,
  last_ping: new Date(),
  created_at: new Date(),
});
